### PR TITLE
Remove deprecations slated for 24.3

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Changes:
- Bumps minimum conda to 23.5.0 (due to incompatibilities with `conda.core.subdir_data.SubdirData._process_raw_repodata`)
- Removes signifiant portion of `conda_build.index` (part of switch to conda-index)
- Removes `conda_build.environ.get_install_actions` in favor of `conda_build.environ.get_package_records` (part of deprecating `conda.models.dist.Dist`)
- More untangling of `conda.exports` and `conda_build.conda_interface`

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: conda_build/environ.py [hunk 1]
```diff
@@ -852,29 +867,35 @@ def package_specs(self):
...
 
-def get_package_records(
-    prefix,
-    specs,
-    env,
-    retries=0,
+# NOTE: The function has to retain the "get_install_actions" name for now since
+#       conda_libmamba_solver.solver.LibMambaSolver._called_from_conda_build
+#       checks for this name in the call stack explicitly.
+def get_install_actions(
```
Comment (kenodegard) [inline conda_build/environ.py:880]: cc @jaimergp @jezdez @mbargull

File: conda_build/cli/main_render.py [hunk 2]
```diff
@@ -193,7 +192,6 @@ def parse_args(args: Sequence[str] | None) -> tuple[ArgumentParser, Namespace]:
     return parser, parser.parse_args(args)
 
 
-@deprecated.argument("24.1.1", "24.3.0", "print_results")
```
Comment (jezdez) [inline conda_build/cli/main_render.py:196]: Not sure I follow with this

File: conda_build/inspect_pkg.py [hunk 3]
```diff
@@ -96,9 +95,6 @@ def __str__(self):
 untracked_package = _untracked_package()
 
 
-@deprecated.argument("24.1.0", "24.3.0", "platform", rename="subdir")
-@deprecated.argument("24.1.0", "24.3.0", "prepend")
-@deprecated.argument("24.1.0", "24.3.0", "minimal_hint")
```
Comment (jezdez) [inline conda_build/inspect_pkg.py:101]: The removal of the arguments had already happened?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -30,7 +30,7 @@ requirements:
   run:
     - beautifulsoup4
     - chardet
-    - conda >=22.11.0
+    - conda >=23.5.0
```
Comment (jezdez) [inline recipe/meta.yaml:33]: NICE!

**Review by jezdez — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

File: conda_build/cli/main_render.py (hunk 2 above)
Comment (kenodegard) [reply conda_build/cli/main_render.py:196]: This was a keyword argument that had been removed, the decorator only raises a warning if the argument was passed in

File: conda_build/inspect_pkg.py (hunk 3 above)
Comment (kenodegard) [reply conda_build/inspect_pkg.py:101]: Correct, the decorator only raises a warning if the argument was passed in

File: conda_build/index.py [hunk 5]
```diff
@@ -325,1419 +243,3 @@ def _delegated_update_index(
...
-    for fn in instructions.get("remove", ()):
-        for key in ("packages", "packages.conda"):
-            if fn.endswith(CONDA_PACKAGE_EXTENSION_V1) and key == "packages.conda":
-                fn = fn.replace(CONDA_PACKAGE_EXTENSION_V1, CONDA_PACKAGE_EXTENSION_V2)
-            popped = repodata[key].pop(fn, None)
-            if popped:
-                repodata["removed"].append(fn)
-    repodata["removed"].sort()
-
-    return repodata
```
[the remark this answers is not in the document]
Comment (mbargull) [reply conda_build/index.py:475]: Addressing this in gh-5226 .

[the remark this answers is not in the document]
Comment (kenodegard) [reply conda_build/index.py:475]: Whoops!

**Pull request merged**